### ISSUE TITLE
Issues with message >  999 bytes?

### ISSUE THREAD

**Issue by winbatch** [adobe-research/libkafka#4]
Using the producer sample, the code barfs if the message I send is 1000 bytes or larger.  I noticed in the code lots of hard coded 1024 values.  Is this the cause?

**Comment by winbatch** [comment]
After I increased the DEFAULT_BUFFER_SIZE fields in Connection.h and Packet.h the problem got fixed.  However, shouldn't this be dynamically allocated???

**Comment by DavidTompkins** [comment]
Greetings,

That buffer is statically allocated as an efficiency mechanism based on
profiling of the library in action. If your use case for kafka uses
variable-sized messages with a wide distribution, then you probably want to
implement a malloc pool.

Related to this, an upcoming update to libkafka will allow this buffer size
to be passed in as a parameter so that you can specify this as part of your
message creation code.

Best of luck.

-DT

**Comment by winbatch** [comment]
Yes, that would be better, especially since messages I might send could
theoretically be binary/not null terminated - so being able to pass the
length would help with that as well.

**Comment by ShikhaSrivastava** [comment]
Hi,

I have been working on libkafka for a while now and I discovered this issue and saw that you mentioned that it will be there in the next release. I implemented a dynamic buffer allocation solution to this where the buffer size is passed as an argument to the ProduceRequest api call. And the calculation of the buffer size takes into consideration the size of each message in the messageArray.
I tested it with more than 10,00,000 google protocol buffer messages and it works fine.

I'd be happy to share the code with the community so that it can be improved further and can be tested to its extremes.

Thanks,
Shikha
